src/components/IDForm/IDForm.js

```diff
@@ -1,23 +1,56 @@
 import './IDForm.css';
-
+import {useEffect} from 'react'
+import { useForm } from 'react-hook-form';
+import {addID} from '../../actions/index'
+import {useDispatch,useSelector} from 'react-redux'
 
 const IDForm = () => {
+  useEffect(() => {
+    //makeApi call
+    //updateState
+  });
+
+  const { handleSubmit, register, errors } = useForm();
+  let dispatch = useDispatch()
+  const onSubmit = values => {
+    dispatch(addID(values))
+  }
+  const countries = useSelector(state => state.countries)
+
+
+  
+  // if(!countries){
+  //   return <h1>Loading</h1>
+  // }
   return (
-    <form>
-      <label for='first-name'>first name</label>
-      <input name='first-name' placeholder='first name'/>
-      <label for='first-name'>last name</label>
-      <input name='last-name' placeholder='last name'/>
-      <label for='country'>target country</label>
-      <input list='countries' name='country' id='countries'/>
-      <datalist id='countries'>
+    
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <input
+        name="first-name"
+        ref={register({
+          required: "Required",
+        })}
+      />
 
+      <input
+        name="last-name"
+        ref={register({
+          required: "Required",
+        })}
+      />
+      <input name="country "type="text" list="data" />
+      <datalist 
+       ref={register({
```

Comment:
I see this throughout the form. Does this mean this part of the input field is required? Either way, great job!